docs/dev/table/streaming/match_recognize.md

```diff
@@ -0,0 +1,654 @@
+
+There are five different strategies:
+* `SKIP TO NEXT ROW` - continues searching for new match starting at the next element to the starting element of a match
+* `SKIP PAST LAST ROW` - resumes pattern matching at the next row after the last row of the current match.
+* `SKIP TO FIRST variable` - resume pattern matching at the first row that is mapped to the pattern variable
+* `SKIP TO LAST variable` - resume pattern matching at the last row that is mapped to the pattern variable
+
+This is also a way to specify to how many matches a single event can belong, e.g. with the `SKIP PAST LAST ROW` each event can belong to at most one match.
+
+To better understand differences between those strategies one can analyze following example:
+
+Query:
+
+{% highlight sql %}
+SELECT * 
+FROM Ticker
+    MATCH_RECOGNIZE(
+        PARTITION BY symbol
+        ORDER BY rowtime
+        MEASURES
+            SUM(A.price) as sumPrice,
+            FIRST(A.rowtime) as startTime,
+            LAST(A.rowtime) as endTime
+        PATTERN (A+ C)
+        ONE ROW PER MATCH
+        [AFTER MATCH STRATEGY]
+        DEFINE
+            A as SUM(A.price) < 30
+    )
+{% endhighlight %}
+
+For input:
+
+{% highlight text %}
+ symbol   tax   price         rowtime
+======== ===== ======= =====================
+ XYZ      1     7       2018-09-17 10:00:01
+ XYZ      2     9       2018-09-17 10:00:02
+ XYZ      1     10      2018-09-17 10:00:03
+ XYZ      2     17      2018-09-17 10:00:04
+ XYZ      2     14      2018-09-17 10:00:05
+{% endhighlight %}
+
+Will produce results based on used `AFTER MATCH STRATEGY`:
+
+* AFTER MATCH SKIP PAST LAST ROW
+
+{% highlight text %}
+ symbol   sumPrice        startTime              endTime
+======== ========== ===================== =====================
+ XYZ      26         2018-09-17 10:00:01   2018-09-17 10:00:03
+ XYZ      17         2018-09-17 10:00:04   2018-09-17 10:00:04
+{% endhighlight %}
+
+* AFTER MATCH SKIP TO LAST A
+
+{% highlight text %}
+ symbol   sumPrice        startTime              endTime
+======== ========== ===================== =====================
+ XYZ      26         2018-09-17 10:00:01   2018-09-17 10:00:03
+ XYZ      27         2018-09-17 10:00:03   2018-09-17 10:00:04
+ XYZ      17         2018-09-17 10:00:04   2018-09-17 10:00:04
+{% endhighlight %}
+
+* AFTER MATCH SKIP TO NEXT ROW
+
+{% highlight text %}
+ symbol   sumPrice        startTime              endTime
+======== ========== ===================== =====================
+ XYZ      26         2018-09-17 10:00:01   2018-09-17 10:00:03
+ XYZ      19         2018-09-17 10:00:02   2018-09-17 10:00:03
+ XYZ      27         2018-09-17 10:00:03   2018-09-17 10:00:04
+ XYZ      17         2018-09-17 10:00:04   2018-09-17 10:00:04
+{% endhighlight %}
+
+* AFTER MATCH SKIP TO FIRST A
+
+This combination will produce runtime exception, because one would try to always start new match where the
```

Comment:
In general for this document, I would not use the word exception, but simply "is not supported".

Reply:
I wanted it to stand out from a situation when we throw exception while compiling query.
This can be compiled, but will throw exception during execution.

Reply:
If your think we need to distinguish between runtime and non-runtime exceptions, we can leave it as it is. I was just wondering how often `exception` occurs in this document. It is obvious that unsupported things throw exceptions at some point.